### PR TITLE
Configure NonceManager for chain reader and CCIPReader.

### PR DESCRIPTION
**Pull request by winder**
* Unit test for CCIPReader Nonces function.
* Configure NonceManager in the chain reader.

### CODE REVIEW

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 1]
```diff
@@ -142,40 +144,53 @@ var SourceReaderConfig = evmrelaytypes.ChainReaderConfig{
 			Configs: map[string]*evmrelaytypes.ChainReaderDefinition{
 				// TODO: update with the consts from https://github.com/smartcontractkit/chainlink-ccip/pull/39
 				// in a followup.
```
Comment (makramkd) [inline core/capabilities/ccip/configs/evm/contract_reader.go]: Can remove the TODO

File: core/capabilities/ccip/configs/evm/contract_reader.go [hunk 2]
```diff
@@ -142,40 +144,53 @@ var SourceReaderConfig = evmrelaytypes.ChainReaderConfig{
...
 					ReadType:          evmrelaytypes.Method,
 				},
-				"GetFeeTokens": {
+				consts.MethodNameGetFeeTokens: {
 					ChainSpecificName: mustGetMethodName("getFeeTokens", priceRegistryABI),
 					ReadType:          evmrelaytypes.Method,
 				},
 			},
 		},
+		consts.ContractNameNonceManager: {
```
Comment (makramkd) [inline core/capabilities/ccip/configs/evm/contract_reader.go:180]: Bump chainlink-ccip to get these consts

File: go.mod [hunk 3]
```diff
@@ -74,7 +74,7 @@ require (
 	github.com/shopspring/decimal v1.4.0
 	github.com/smartcontractkit/chain-selectors v1.0.21
 	github.com/smartcontractkit/chainlink-automation v1.0.4
-	github.com/smartcontractkit/chainlink-ccip v0.0.0-20240827164549-33f5819d7ddc
+	github.com/smartcontractkit/chainlink-ccip v0.0.0-20240828135226-207ea3027224
```
Comment (winder) [inline go.mod]: TODO: switch this to ccip-develop once [the branch has been merged](https://github.com/smartcontractkit/chainlink-ccip/pull/77).

File: contracts/src/v0.8/ccip/test/helpers/CCIPReaderTester.sol [hunk 4]
```diff
@@ -9,6 +9,7 @@ contract CCIPReaderTester {
 
   mapping(uint64 sourceChainSelector => OffRamp.SourceChainConfig sourceChainConfig) internal s_sourceChainConfigs;
   mapping(uint64 destChainSelector => uint64 sequenceNumber) internal s_destChainSeqNrs;
+  mapping(uint64 sourceChainSelector => mapping(bytes sender => uint64 nonce)) internal s_senderNonce;
```
Comment (makramkd) [inline contracts/src/v0.8/ccip/test/helpers/CCIPReaderTester.sol:12]: I think in this particular instance we can actually just deploy the NonceManager contract directly - it doesn't seem to have any dependents and is basically standalone. You only have to reference the authorized callers which would be the address of the private key generated in the test.

The advantage of this is we're testing against the same contract that is going to be deployed in prod so any breakages in ABI there will end up breaking the test, forcing us to fix it.

**Review by makramkd — APPROVED** [review]

**Review by dimkouv — APPROVED** [review]

**Pull request closed without merging**